### PR TITLE
Tech 1585 org unit ou mode in request

### PR DESCRIPTION
**Pull request by muilpp**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java [hunk 1]
```diff
@@ -306,8 +305,12 @@ private String getOrgUnitModeViolation(EventOperationParams params, User user, P
       case ALL -> userCanSearchOrgUnitModeALL(user)
           ? null
           : "Current user is not authorized to query across all organisation units";
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java:307]: This validation altogether couldn't be moved to EventRequestParamsMapper?
For most of the modes it is just checking existence or not of orgUnits, for ALL mode we are checking the user and it shouldn't access the DB

Comment (muilpp) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java:307]: `ACCESSIBLE` and `CAPTURE` need the user to validate the org unit mode, and I wanted to keep all the unit mode validations together, that's the reason why they are in this mapper now.

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java [hunk 2]
```diff
@@ -285,9 +284,7 @@ private String getOrgUnitModeViolation(EventOperationParams params, User user, P
...
       case ACCESSIBLE -> getAccessibleScopeValidation(user, program);
       case CAPTURE -> getCaptureScopeValidation(user);
-      case CHILDREN, SELECTED, DESCENDANTS -> params.getOrgUnitUid() == null
-          ? "Organisation unit is required for orgUnitMode: " + params.getOrgUnitMode()
-          : null;
+      default -> null;
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java:287]: Shouldn't we have here now the validation for `ALL` mode?

userCanSearchOuModeALL(user)

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapper.java [hunk 3]
```diff
@@ -225,15 +231,23 @@ private void validateUpdateDurationParams(RequestParams requestParams)
     }
   }
 
-  // TODO Use RequestParamUtils.validateOrgUnitParams as soon /events accepts a list of org units
-  // UIDs
-  private void validateOrgUnitParams(UID orgUnit, OrganisationUnitSelectionMode orgUnitMode)
-      throws BadRequestException {
-    if (orgUnit != null && (orgUnitMode == ACCESSIBLE || orgUnitMode == CAPTURE)) {
+  private void validateOrgUnitMode(RequestParams params) throws BadRequestException {
+    if ((params.getOrgUnitMode().equals(ACCESSIBLE) || params.getOrgUnitMode().equals(CAPTURE))
```
Comment (enricocolasante) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapper.java:235]: `params.getOrgUnitMode()` could be null at this point?
If yes, we could have an NPE here

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java [hunk 4]
```diff
@@ -112,6 +112,7 @@ public EventSearchParams map(EventOperationParams operationParams)
     ProgramStage programStage = validateProgramStage(operationParams.getProgramStageUid());
     OrganisationUnit requestedOrgUnit = validateRequestedOrgUnit(operationParams.getOrgUnitUid());
 
+    validateOrgUnitMode(operationParams, user, program);
     OrganisationUnitSelectionMode orgUnitMode =
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java:116]: This default assignment of orgUnitMode in case something is null shouldn't be called before validation of the orgUnitMode?

What if the default assignment become `ACCESSIBLE` but the user has no orgUnits in search scope?

In my opinion we could move `OrganisationUnitSelectionMode orgUnitMode = getOrgUnitMode(requestedOrgUnit, operationParams.getOrgUnitMode());` in `RequestParamsMapper`, that is generally the place where we assign default values anyway

**Review by enricocolasante — COMMENTED** [review]
A couple of doubts before :heavy_check_mark:

File: dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapperTest.java [hunk 5]
```diff
@@ -483,6 +481,8 @@ void shouldFailWhenOrgUnitSuppliedAndOrgUnitModeAccessible() {
     requestParams.setOrgUnit(UID.of(orgUnit.getUid()));
     requestParams.setOrgUnitMode(ACCESSIBLE);
 
+    when(organisationUnitService.getOrganisationUnit(orgUnit.getUid())).thenReturn(orgUnit);
```
Comment (teleivo) [inline dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapperTest.java:484]: nit: I think we put the parameter to the method/function we test close to the call of the tested method. So the mock would go above the `requestParams` statements.

File: dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapperTest.java [hunk 6]
```diff
@@ -495,42 +495,40 @@ void shouldFailWhenOrgUnitSuppliedAndOrgUnitModeCapture() {
...
     RequestParams requestParams = new RequestParams();
-    requestParams.setOrgUnit(UID.of(orgUnit.getUid()));
-    requestParams.setOrgUnitMode(SELECTED);
+    requestParams.setOrgUnitMode(OrganisationUnitSelectionMode.SELECTED);
 
-    EventOperationParams eventOperationParams = mapper.map(requestParams);
+    Exception exception = assertThrows(BadRequestException.class, () -> mapper.map(requestParams));
 
-    assertEquals(SELECTED, eventOperationParams.getOrgUnitMode());
+    assertStartsWith("orgUnit is required for orgUnitMode: SELECTED", exception.getMessage());
```
Comment (teleivo) [inline dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapperTest.java:512]: Might be worth making one parameterized test for these 3 cases. Something like

```java
@ParameterizedTest
@EnumSource(value = OrgUnitMode.class, names = {"SELECTED", "DESCENDANTS", "CHILDREN"})
```

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java (hunk 2 above)
Comment (muilpp) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java:287]: that's in the line 282, I can't put it here because CHILDREN, SELECTED and DESCENDANTS would be affected.

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java (hunk 4 above)
Comment (muilpp) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java:116]: Nice catch!

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapper.java (hunk 3 above)
Comment (muilpp) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapper.java:235]: there's a condition on this method call, it's just called if orgUnitMode is not null.

File: dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapperTest.java (hunk 5 above)
Comment (muilpp) [reply dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapperTest.java:484]: my memory is really bad lately, so I could have easily have forgotten it, when did we decide this?

Comment (teleivo) [reply dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapperTest.java:484]: I don't think we made a decision. Its just what I see us doing in all the mapper tests AFAIK 😅

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java [hunk 7]
```diff
@@ -112,14 +112,15 @@ public EventSearchParams map(EventOperationParams operationParams)
     ProgramStage programStage = validateProgramStage(operationParams.getProgramStageUid());
     OrganisationUnit requestedOrgUnit = validateRequestedOrgUnit(operationParams.getOrgUnitUid());
 
-    validateOrgUnitMode(operationParams, user, program);
-    OrganisationUnitSelectionMode orgUnitMode =
-        getOrgUnitMode(requestedOrgUnit, operationParams.getOrgUnitMode());
+    if (operationParams.getOrgUnitMode() != null) {
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java:115]: So far most validation functions get the `parameter` passed in regardless of whether its null or not. They then start with a guard clause exiting when there is nothing to validate. This particular function might already do a null check.

This makes the code in the main mapper function more readable IMHO.

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java [hunk 8]
```diff
@@ -133,22 +133,26 @@ class EventOperationParamsMapperTest {
...
 
   @Test
   void shouldFailWithForbiddenExceptionWhenUserHasNoAccessToProgramStage() {
     programStage = new ProgramStage();
     programStage.setUid("PlZSBEN7iZd");
     EventOperationParams eventOperationParams =
-        EventOperationParams.builder().programStageUid(programStage.getUid()).build();
+        EventOperationParams.builder()
+            .programStageUid(programStage.getUid())
+            .orgUnitMode(ACCESSIBLE)
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java:154]: Do we need to set this as it cannot be null and this test is testing the mapper. So the mapping in the web has not happened which would set the default?

Would it still make sense to add a default to the builder?

File: dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java [hunk 9]
```diff
@@ -369,11 +373,37 @@ void shouldReturnSelectedOrgUnitEventsWhenNoOuModeSpecifiedAndUserHasAccessToOrg
                     + " instead"));
   }
 
+  @Test
+  void shouldReturnAccessibleOrgUnitEventsWhenNoOuModeNorOrgUnitSpecifiedAndUserHasAccessToOrgUnit()
```
Comment (teleivo) [inline dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java:377]: Can this name be simplified to

```suggestion
  void shouldReturnAccessibleOrgUnitEvents()
```

the exact params are visible from the builder

File: dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java [hunk 10]
```diff
@@ -369,11 +373,37 @@ void shouldReturnSelectedOrgUnitEventsWhenNoOuModeSpecifiedAndUserHasAccessToOrg
...
+
   @Test
   void shouldFailWhenNoOuModeSpecifiedAndUserHasNoAccessToOrgUnit() {
     injectSecurityContext(userService.getUser("FIgVWzUCkpw"));
     EventOperationParams params =
-        EventOperationParams.builder().programUid("pcxIanBWlSY").orgUnitUid("DiszpKrYNg8").build();
+        EventOperationParams.builder()
+            .programUid("pcxIanBWlSY")
+            .orgUnitUid("DiszpKrYNg8")
+            .orgUnitMode(SELECTED)
```
Comment (teleivo) [inline dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java:405]: The test name is 

`shouldFailWhenNoOuModeSpecifiedAndUserHasNoAccessToOrgUnit`

should probably be something like

`shouldFailWhenUserHasNoAccessToSelectedOrgUnit`

File: dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java [hunk 11]
```diff
@@ -384,7 +414,8 @@ void shouldFailWhenNoOuModeSpecifiedAndUserHasNoAccessToOrgUnit() {
   void shouldReturnAccessibleOrgUnitEventsWhenNoOrgUnitSpecifiedNorOuModeSpecified()
       throws ForbiddenException, BadRequestException {
     injectSecurityContext(userService.getUser("FIgVWzUCkpw"));
-    EventOperationParams params = EventOperationParams.builder().programUid("pcxIanBWlSY").build();
+    EventOperationParams params =
+        EventOperationParams.builder().programUid("pcxIanBWlSY").orgUnitMode(ACCESSIBLE).build();
```
Comment (teleivo) [inline dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java:418]: The test is called

`shouldFailWhenNoOuModeSpecifiedAndUserHasNoAccessToOrgUnit`

and should probably be called

`shouldFailWhenUserHasNoAccessToOrgUnit` or `shouldFailWhenUserHasNoAccessToOrgUnitGivenOrgUnitModeAccessible`

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapper.java [hunk 12]
```diff
@@ -68,11 +68,16 @@ class EventRequestParamsMapper {
...
+            requestParams.getOrgUnit(),
+            validateDeprecatedParameter(
+                "ouMode",
+                requestParams.getOuMode(),
+                "orgUnitMode",
+                requestParams.getOrgUnitMode()));
 
-    if (requestParams.getOrgUnitMode() != null) {
-      validateOrgUnitMode(requestParams);
+    if (orgUnitMode != null) {
```
Comment (teleivo) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapper.java:79]: Same comment as in EventOperationParamsMapper `validateOrgUnitMode`

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapper.java [hunk 13]
```diff
@@ -231,23 +236,39 @@ private void validateUpdateDurationParams(RequestParams requestParams)
     }
   }
 
-  private void validateOrgUnitMode(RequestParams params) throws BadRequestException {
-    if ((params.getOrgUnitMode().equals(ACCESSIBLE) || params.getOrgUnitMode().equals(CAPTURE))
-        && params.getOrgUnit() != null) {
+  private void validateOrgUnitMode(UID orgUnit, OrganisationUnitSelectionMode orgUnitMode)
+      throws BadRequestException {
+    if ((orgUnitMode.equals(ACCESSIBLE) || orgUnitMode.equals(CAPTURE)) && orgUnit != null) {
```
Comment (teleivo) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapper.java:241]: We should compare enums using `==` instead of `.equals` as `==` is null-safe. Even if you already checked that the mode cannot be null its a safer way to compare enums in general.

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java (hunk 7 above)
Comment (muilpp) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java:115]: Right. This check will be done in the web layer from now on, so here won't be necessary anymore because it can't be null at this point.

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java (hunk 8 above)
Comment (muilpp) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java:154]: Yes, that's the reason why I need to set the org unit mode in each particular test.

File: dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java (hunk 9 above)
Comment (muilpp) [reply dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java:377]: I guess that's just my personal taste, but I like having the important details of the test in the title, so just by reading it I know what to expect.

File: dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java (hunk 10 above)
Comment (muilpp) [reply dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java:405]: Yeah. Anyway, with the latest changes this test is not useful anymore.

File: dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java (hunk 11 above)
Comment (muilpp) [reply dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java:418]: Same as above, this test will be deleted, it has no value anymore.

File: dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java (hunk 9 above)
Comment (teleivo) [reply dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java:377]: Yes probably subjective but to me the essence is getting the accessible org units 😅  Reapeating most params seems to redundant/verbose IMHO

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapper.java [hunk 14]
```diff
@@ -189,15 +198,39 @@ private void validateUpdateDurationParams(RequestParams requestParams)
...
+    if ((orgUnitMode == ACCESSIBLE || orgUnitMode == CAPTURE) && orgUnit != null) {
       throw new BadRequestException(
           String.format(
               "orgUnitMode %s cannot be used with orgUnits. Please remove the orgUnit parameter and try again.",
               orgUnitMode));
     }
+
+    if ((orgUnitMode.equals(CHILDREN)
+            || orgUnitMode.equals(SELECTED)
+            || orgUnitMode.equals(DESCENDANTS))
```
Comment (enricocolasante) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapper.java:212]:
```suggestion
    if ((orgUnitMode == CHILDREN
            || orgUnitMode == SELECTED
            || orgUnitMode == DESCENDANTS)
```

**Review by enricocolasante — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java [hunk 15]
```diff
@@ -253,28 +251,25 @@ private void validateAttributeOptionCombo(CategoryOptionCombo attributeOptionCom
...
+  private void validateOrgUnitMode(
+      OrganisationUnitSelectionMode orgUnitMode, User user, Program program)
       throws BadRequestException {
-    if (params.getOrgUnitMode() != null) {
-      String violation = getOrgUnitModeViolation(params, user, program);
-      if (violation != null) {
-        throw new BadRequestException(violation);
-      }
+
+    String violation = getOrgUnitModeViolation(orgUnitMode, user, program);
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java:258]: Can we inline this method `getOrgUnitModeViolation`? I think its only used once

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java [hunk 16]
```diff
@@ -303,6 +305,7 @@ void shouldMapAttributeFilters() throws BadRequestException, ForbiddenException
                     List.of(new QueryFilter(QueryOperator.EQ, "2")),
                     TEA_2_UID,
                     List.of(new QueryFilter(QueryOperator.LIKE, "foo"))))
+            .orgUnitMode(SELECTED)
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java:308]: How can it be selected when there is no orgUnit?

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java [hunk 17]
```diff
@@ -686,14 +685,8 @@ void shouldMapSearchScopeOrgUnitWhenOuModeAccessible()
     user.setOrganisationUnits(Set.of(searchScopeOrgUnit));
 
     when(currentUserService.getCurrentUser()).thenReturn(user);
-    when(organisationUnitService.getOrganisationUnit(orgUnit.getUid())).thenReturn(orgUnit);
 
-    EventOperationParams requestParams =
-        EventOperationParams.builder()
-            .programUid(program.getUid())
-            .orgUnitUid(orgUnit.getUid())
-            .orgUnitMode(ACCESSIBLE)
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java:695]: Do we default to mode accessible? You removed it and the test says `shouldMapSearchScopeOrgUnitWhenOuModeAccessible`

Might have been my mistake at some point 🤷🏻 I see the use of `requestParams` name while this class should be `operationparams`. We could replace all with `params` in all the mapper tests then there is no confusion 😅

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java [hunk 18]
```diff
@@ -746,6 +739,7 @@ void shouldMapRequestedOrgUnitAsSelectedWhenOrgUnitProvidedAndNoOrgUnitModeProvi
         EventOperationParams.builder()
             .programUid(program.getUid())
             .orgUnitUid(orgUnit.getUid())
+            .orgUnitMode(SELECTED)
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java:742]: The test name does not make sense I think

`shouldMapRequestedOrgUnitAsSelectedWhenOrgUnitProvidedAndNoOrgUnitModeProvided`

you do provide a mode now

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java (hunk 15 above)
Comment (muilpp) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java:258]: It's true it's just used once, I just followed this mapper's pattern, above there are the methods `validateProgram`, `validateProgramStage` and `validateRequestedOrgUnit`, all of them also just used once.

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java (hunk 16 above)
Comment (muilpp) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java:308]: nice one, this one should be accessible

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java (hunk 17 above)
Comment (muilpp) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java:695]: Yes, in this test case we default to ACCESSIBLE.
Good idea to not use requestParams.

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java (hunk 18 above)
Comment (muilpp) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java:742]: Right, it's another test that has no value after the latest commits

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java (hunk 15 above)
Comment (teleivo) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java:255]: I mean inline method `getOrgUnitModeViolation`. You would still have `validateOrgUnitMode`

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java [hunk 19]
```diff
@@ -131,22 +134,25 @@ class EventOperationParamsMapperTest {
...
     orgUnit.setChildren(
         Set.of(
             createOrgUnit("captureScopeChild", "captureScopeChildUid"),
             createOrgUnit("searchScopeChild", "searchScopeChildUid")));
+
+    user = new User();
+    user.setOrganisationUnits(Set.of(orgUnit));
+    when(currentUserService.getCurrentUser()).thenReturn(user);
+
+    eventBuilder = eventBuilder.orgUnitMode(ACCESSIBLE);
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java:147]: Can you please add a comment why its set to accessible. I am sure I won't know anymore next week 😅

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java (hunk 17 above)
Comment (teleivo) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapperTest.java:695]: Not sure why we set the mode in the setup. It is clearer if its set here IMHO

File: dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/OrderAndPaginationExporterTest.java [hunk 20]
```diff
@@ -107,6 +111,9 @@ void setUp() {
     // needed as some tests are run using another user (injectSecurityContext) while most tests
     // expect to be run by admin
     injectAdminUser();
+
+    operationParamsBuilder = EventOperationParams.builder();
```
Comment (teleivo) [inline dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/OrderAndPaginationExporterTest.java:115]: This test class is to test ordering/pagination of all enitites (right now its TE, EV).

So this should then probably be `eventOperationParamsBuilder`. We can shorten this to `eventParamsBuilder` IMHO

**Review by teleivo — APPROVED** [review]
